internal/api/create_test_plan.go

```diff
@@ -19,6 +19,7 @@ type TestPlanParams struct {
 	Identifier  string             `json:"identifier"`
 	Parallelism int                `json:"parallelism"`
 	Branch      string             `json:"branch"`
+	RetryCount  int                `json:"retryCount"`
```

Comment:
Actually, this value is expected in the fetch test plan endpoint, not to create test plan endpoint. The retried job will not hit the create test plan endpoint because the test plan is already been created when the job is retried. Also, the server expected the value as `job_retry_count`.